extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -203,7 +203,9 @@ public ReturnOrAwait<Object> runIncrementally(IntSet readableInputs)
         final Frame frame = inputChannel.read();
         convertRowFrameToRowsAndColumns(frame);
       } else if (inputChannel.isFinished()) {
-        runAllOpsOnMultipleRac(frameRowsAndCols);
+        if (!frameRowsAndCols.isEmpty()) {
+          runAllOpsOnMultipleRac(frameRowsAndCols);
```

Comment:
Shouldn't the check be inside runAllOpsMulitpleRAC ?

Reply:
Have made the change.